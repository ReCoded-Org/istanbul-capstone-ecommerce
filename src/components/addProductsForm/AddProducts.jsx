import React from "react";

export default function AddProducts() {
  return <div>add Products</div>;
}
