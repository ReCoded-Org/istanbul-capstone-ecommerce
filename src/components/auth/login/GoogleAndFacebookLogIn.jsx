import React from "react";
import db from "../../config/firebaseConfig";
import * as firebase from "firebase";
import "firebase/auth";

export default function GoogleAndFacebookLogIn() {
  const googleProvider = new firebase.auth.GoogleAuthProvider();
  const facebookProvider = new firebase.auth.FacebookAuthProvider();

  const postData = async (user) => {
    const info = {
      name: user.displayName,
      userId: user.uid,
      imageUrl: user.photoURL,
      isAdmin: false,
    };
    const createNewSubCollection = await db
      .collection("users")
      .doc(info.userId)
      .collection("liked")
      .add({
        liked: 0,
      });
    await db.collection("users").doc(info.userId).set(info);
  };

  const logIn = (provider) => {
    firebase
      .auth()
      .signInWithPopup(provider)
      .then((result) => {
        const user = result.user;
        postData(user);
      })
      .catch(function (error) {
        console.log(error.message);
      });
  };

  return (
    <div>
      <button onClick={() => logIn(googleProvider)}>Log in with Google</button>
      <button onClick={() => logIn(facebookProvider)}>
        Log in with Facebook
      </button>
    </div>
  );
}
