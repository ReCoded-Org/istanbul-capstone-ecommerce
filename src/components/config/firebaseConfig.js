import * as firebase from "firebase";
import "firebase/firestore";
import "firebase/database";

var firebaseConfig = {
  apiKey: "AIzaSyCh4oNUuWlrY8QZMGwZ4zOivJYwSi_ezbE",
  authDomain: "testcapstone-project.firebaseapp.com",
  databaseURL: "https://testcapstone-project.firebaseio.com",
  projectId: "testcapstone-project",
  storageBucket: "testcapstone-project.appspot.com",
  messagingSenderId: "866376480749",
  appId: "1:866376480749:web:24063b63ba331fb83a19e4",
};

// Initialize Firebase
firebase.initializeApp(firebaseConfig);
const db = firebase.firestore();
export const auth = firebase.auth();
export default db;
