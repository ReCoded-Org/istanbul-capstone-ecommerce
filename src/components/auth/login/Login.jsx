import React from "react";
import { auth } from "../../config/firebaseConfig";
import db from "../../config/firebaseConfig";
import AddProducts from "../../addProductsForm/AddProducts";
import GoogleAndFacebookLogIn from "./GoogleAndFacebookLogIn";

const LogInForm = () => {
  const [isAdmin, setIsAdmin] = React.useState(false);

  const userLogout = () => {
    auth.signOut().then(() => {
      setIsAdmin(false);
    });
  };
  const userLogin = async (e) => {
    e.preventDefault();
    const email = e.target[0].value;
    const password = e.target[1].value;
    const userLogin = await auth.signInWithEmailAndPassword(email, password);
    const loggedInUserName = await db
      .collection("users")
      .doc(userLogin.user.uid)
      .get();
    setIsAdmin(loggedInUserName.data().isAdmin);
  };

  return (
    <>
      {isAdmin && <AddProducts />}
      <h5>sign in</h5>
      <form onSubmit={userLogin}>
        <input type="email" placeholder="Email" />
        <input type="password" placeholder="Password" />
        <button type="submit">Log in</button>
      </form>

      <button onClick={userLogout}>Logout</button>
      <GoogleAndFacebookLogIn />
    </>
  );
};

export default LogInForm;
