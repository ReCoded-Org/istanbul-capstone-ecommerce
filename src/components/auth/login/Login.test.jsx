import React from "react";
import renderer from "react-test-renderer";
import Login from "./Login";
import { auth } from "../../config/firebaseConfig";
import db from "../../config/firebaseConfig";
import AddProducts from "../../addProductsForm/AddProducts";
import GoogleAndFacebookLogIn from "./GoogleAndFacebookLogIn";

it("renders correctly", () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});
