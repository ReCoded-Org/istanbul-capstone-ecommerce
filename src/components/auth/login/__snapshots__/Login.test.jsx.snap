// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h5>
    sign in
  </h5>,
  <form
    onSubmit={[Function]}
  >
    <input
      id="emailField"
      placeholder="Email"
      type="email"
    />
    <input
      id="passwordField"
      placeholder="Password"
      type="password"
    />
    <button
      type="submit"
    >
      Log in
    </button>
  </form>,
  <button
    onClick={[Function]}
  >
    Logout
  </button>,
  <div>
    <button
      onClick={[Function]}
    >
       Log in with Google
    </button>
    <button
      onClick={[Function]}
    >
       
      Log in with Facebook
    </button>
  </div>,
]
`;
